.cbmc-batch/jobs/aws_cryptosdk_enc_ctx_serialize/README.md

```diff
@@ -0,0 +1,38 @@
+# Memory safety proof for aws_cryptosdk_enc_ctx_serialize
+
+This proof harness attains 95% code coverage.  The following comments explain
+why the uncovered lines of code are unreachable code.
+
+Some functions contain unreachable blocks of code:
+
+* `aws_cryptosdk_enc_ctx_serialize`
+
+    * aws_array_list_get_at_ptr is never given an invalid index
+
+* `hash_table_state_is_valid`
+
+    *  map is never NULL
+
+* `aws_array_list_is_valid`
+
+    *  list is never NULL
```

Comment:
should it be?

Reply:
Not as far as I can tell. The calls to this function are from inside a loop iterating through elements of the aws_array_list (such as from aws_array_list_get_at_ptr), meaning that to reach a call to this function the list is non NULL.